contracts/staking/IStaking.sol

```diff
@@ -0,0 +1,252 @@
+    ///         based on the liquid T staked. Can be called only by the
+    ///         delegation owner and operator.
+    function unstakeNu(address operator, uint256 amount) external;
+
+    /// @notice Sets cached legacy stake amount to 0, sets the liquid T stake
+    ///         amount to 0 and withdraws all liquid T from the stake to the
+    ///         owner. Reverts if there is at least one non-zero authorization.
+    ///         Can be called only by the delegation owner and operator.
+    function unstakeAll(address operator) external;
+
+    //
+    //
+    // Keeping information in sync
+    //
+    //
+
+    /// @notice Notifies about the discrepancy between legacy KEEP active stake
+    ///         and the amount cached in T staking contract. Slashes the operator
+    ///         in case the amount cached is higher than the actual active stake
+    ///         amount in KEEP staking contract. Needs to update authorizations
+    ///         of all affected applications and execute an involuntary
+    ///         allocation decrease on all affected applications. Can be called
+    ///         by anyone, notifier receives a reward.
+    function notifyKeepStakeDiscrepancy(address operator) external;
+
+    /// @notice Notifies about the discrepancy between legacy NU active stake
+    ///         and the amount cached in T staking contract. Slashes the
+    ///         operator in case the amount cached is higher than the actual
+    ///         active stake amount in NU staking contract. Needs to update
+    ///         authorizations of all affected applications and execute an
+    ///         involuntary allocation decrease on all affected applications.
+    ///         Can be called by anyone, notifier receives a reward.
+    function notifyNuStakeDiscrepancy(address operator) external;
+
+    /// @notice Sets the penalty amount for stake discrepancy and reward
+    ///         multiplier for reporting it. The penalty is seized from the
+    ///         operator account, and 5% of the penalty, scaled by the
+    ///         multiplier, is given to the notifier. The rest of the tokens are
+    ///         burned. Can only be called by the Governance. See `seize` function.
+    function setStakeDiscrepancyPenalty(
+        uint256 penalty,
+        uint256 rewardMultiplier
+    ) external;
+
+    /// @notice Adds operators to the slashing queue along with the amount that
+    ///         should be slashed from each one of them. Can only be called by
+    ///         application authorized for all operators in the array.
+    function slash(uint256 amount, address[] memory operators) external;
+
+    /// @notice Adds operators to the slashing queue along with the amount,
+    ///         reward multiplier and notifier address. The notifier will
+    ///         receive 1% of the slashed amount scaled by the reward adjustment
+    ///         parameter once the seize order will be processed. Can only be
```

Comment:
We need to check the stake of each operator to ensure that we can do this safely; it is possible that a naive implementation would result in an inconsistent state where the staking contract doesn't maintain full reserves.